### PR TITLE
Add support for mathematical operations in values

### PR DESCRIPTION
**Pull request by XGHeaven**
I implement it #106 

But just implement `+` `*` `-` `/`, without `()`

Changes:
Modify:
    - move number to _number

Add:
    - new number parse
    - space function like white except skip comment
    - isOperator function to check if char is math operator
    - operatorPrivilege function to compare privilege with op1 and op2
    - doMathExpression function to eval result based on operator

Any Suggestion Welcome...

### CODE REVIEW

**Comment by jordanbtucker** [comment]
This look great at a glance. I don't have time to vet this in depth right now, but I'll definitely take a look later. For now, I'll just leave a few comments in the code. Thanks for the great work!

File: lib/json5.js [hunk 1]
```diff
@@ -124,9 +125,9 @@ JSON5.parse = (function () {
             return key;
         },
 
-        number = function () {
+        _number = function () {
```
Comment (jordanbtucker) [inline lib/json5.js]: I'd like to use a better name than `_number` here. Let's come up with a names that describe the difference between `number()` and `_number()`.

File: lib/json5.js [hunk 2]
```diff
@@ -350,6 +396,17 @@ JSON5.parse = (function () {
             }
         },
 
+        space = function () {
+// like `white`, but this only skip whitespace
+            while(ch) {
+                if (~ws.indexOf(ch)) {
```
Comment (jordanbtucker) [inline lib/json5.js]: I know that `~-1 == 0 == false` but I don't see this used very often, so let's simplify this to `ws.indexOf(ch) >= 0` for readability.

File: lib/json5.js [hunk 3]
```diff
@@ -393,6 +450,36 @@ JSON5.parse = (function () {
             error("Unexpected " + renderChar(ch));
         },
 
+        isOperator = function(ch) {
+            return !!~operator.indexOf(ch);
```
Comment (jordanbtucker) [inline lib/json5.js]: Again, let's use `operator.indexOf(ch) >= 0`.

**Comment by jordanbtucker** [comment]
I'd like to see some test cases using other number literals like `NaN` and `Infinity` and for cases like dividing by zero (`1 / 0`) and overflowing the operation with large numbers.

File: lib/json5.js (hunk 1 above)
Comment (XGHeaven) [inline lib/json5.js]: How about `numberLiteral` instead of `_number`?

File: lib/json5.js (hunk 2 above)
Comment (XGHeaven) [inline lib/json5.js]: Ok...

**Comment by XGHeaven** [comment]
@jordanbtucker I change some code.

File: lib/json5.js [hunk 4]
```diff
@@ -393,6 +450,36 @@ JSON5.parse = (function () {
             error("Unexpected " + renderChar(ch));
         },
 
+        isOperator = function(ch) {
+            return operator.indexOf(ch) >= 0;
+        },
+
+        // return op1 > op2
+        operatorPrivilege = function(op1, op2) {
```
Comment (amb26) [inline lib/json5.js]: More usual term here is "Precedence"

File: lib/json5.js [hunk 5]
```diff
@@ -43,6 +43,7 @@ JSON5.parse = (function () {
             '\xA0',
             '\uFEFF'
         ],
+        operator = ['+', '-', '*', '/'],
```
Comment (amb26) [inline lib/json5.js]: Why not support all operators which could legitimately operate on numeric constants - therefore including !, ^, &, &&, |, and ||. I think there's no reason to exclude boolean constants from this work - therefore supporting true/false in addition to the numbers which have been already supported.

Comment (amb26) [inline lib/json5.js]: Support for arbitrary grouping using () would also round out this feature and make it more useful.

**Comment by jordanbtucker** [comment]
Hey, everyone. I'm going to close this issue because it is incompatible with the [official specification](https://json5.github.io/json5-spec/). If you would like to continue this discussion, please open an issue on the [official specification repository](https://github.com/json5/json5-spec). Please see the new [issues template](https://github.com/json5/json5/blob/master/.github/issue_template.md) for more information. Thanks.

**Pull request closed without merging**